functions/src/artisan.functions.ts

```diff
@@ -0,0 +1,51 @@
+import * as admin from 'firebase-admin'
+
+export async function addArtisan(data) {
+  console.log("Adding artisan with the following data:")
+  console.log(data)
+
+  let artisanInfo = data.artisanInfo
+  let cgaID = data.cgaID
+
+  let artisanDBInfo = {
+    cgaID: cgaID,
+    ...artisanInfo
+  }
+
+  // Add artisan to artisans root
+  let newArtisan = admin.database().ref('artisans/').push(artisanDBInfo)
+
+  // Add phone to artisan ID to phoneMap root
+  await admin.database()
+    .ref(`phoneMap/${artisanInfo.phoneNumber}`)
+    .set(newArtisan.key)
+
+  var artisanObject = {
+    name: artisanInfo.name,
+    phoneNumber: artisanInfo.phoneNumber,
+    description: artisanInfo.description,
+    uid: newArtisan.key
+  }
+
+  // Add artisanID to cga in amazonUsers root
+  admin.database().ref(`amazonUsers/${cgaID}/artisans/${newArtisan.key}`).set(true)
+
+  console.log("Returning:")
+  console.log(artisanObject)
+
+  return artisanObject
+}
+
+export async function deleteArtisan(data) {
+  console.log("Deleting artisan with the following data:")
+  console.log(data)
+
+  let artisanID = data.uid
+  let artisanInfo = (await admin.database().ref(`artisans/${artisanID}`).once('value')).val()
+  
+  let artisanRootTask = admin.database().ref(`artisans/${artisanID}`).remove()
```

Comment:
You delete the whole artisan and then try to use the artisan fields in the remaining deletes. This might need to be the last deletion.

Reply:
The artisan fields are stored in artisanInfo. Just because I call remove on the database in the cloud, doesnt mean I don't have access to the info locally still. So it should work fine as is.